Agent fails when using log4j2 without slf4j-api on the classpath

## Describe the bug
Using log4j2 (or log4j) without slf4j-api on the classpath crashes the agent.

## Steps to reproduce
```java
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class Main {

    private static final Logger LOGGER = LogManager.getLogger("Main");

    public static void main(String[] args) {
        LOGGER.info("foo");
    }
}
```

`java -javaagent:.../elastic-apm-agent-1.22.0.jar -Delastic.apm.application_packages=Main -cp ... Main`

## Expected behavior
```
2021-03-29 19:42:31,588 INFO  [main] [{}] Main - foo
```

## Actual Behavior
```
java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
	at co.elastic.apm.agent.log4j2.Log4j2LogShadingHelper.<clinit>(Log4j2LogShadingHelper.java:46)
	at co.elastic.apm.agent.log4j2.Log4j2AppenderAppendAdvice.shadeAndSkipIfOverrideEnabled(Log4j2AppenderAppendAdvice.java:38)
	at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.append(AbstractOutputStreamAppender.java:181)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:456)
	at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:82)
	at org.apache.logging.log4j.core.Logger.log(Logger.java:161)
	at org.apache.logging.log4j.spi.AbstractLogger.tryLogMessage(AbstractLogger.java:2205)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessageTrackRecursion(AbstractLogger.java:2159)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessageSafely(AbstractLogger.java:2142)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessage(AbstractLogger.java:2017)
	at org.apache.logging.log4j.spi.AbstractLogger.logIfEnabled(AbstractLogger.java:1983)
	at org.apache.logging.log4j.spi.AbstractLogger.info(AbstractLogger.java:1320)
	at Main.main(Main.java:11)
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
	at co.elastic.apm.agent.shaded.bytebuddy.dynamic.loading.ByteArrayClassLoader.findClass(ByteArrayClassLoader.java:369)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at co.elastic.apm.agent.shaded.bytebuddy.dynamic.loading.ByteArrayClassLoader$ChildFirst.loadClass(ByteArrayClassLoader.java:1229)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 20 more
java.lang.NoClassDefFoundError: Could not initialize class co.elastic.apm.agent.log4j2.Log4j2LogShadingHelper
	at co.elastic.apm.agent.log4j2.Log4j2AppenderAppendAdvice.shadeLoggingEvent(Log4j2AppenderAppendAdvice.java:46)
	at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.append(AbstractOutputStreamAppender.java:186)
	at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
	at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
	at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:540)
	at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:498)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:481)
	at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:456)
	at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:82)
	at org.apache.logging.log4j.core.Logger.log(Logger.java:161)
	at org.apache.logging.log4j.spi.AbstractLogger.tryLogMessage(AbstractLogger.java:2205)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessageTrackRecursion(AbstractLogger.java:2159)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessageSafely(AbstractLogger.java:2142)
	at org.apache.logging.log4j.spi.AbstractLogger.logMessage(AbstractLogger.java:2017)
	at org.apache.logging.log4j.spi.AbstractLogger.logIfEnabled(AbstractLogger.java:1983)
	at org.apache.logging.log4j.spi.AbstractLogger.info(AbstractLogger.java:1320)
	at Main.main(Main.java:11)
```

--- a/CHANGELOG.asciidoc
+++ b/CHANGELOG.asciidoc
@@ -62,7 +62,8 @@ Even when matching on the main class name or on system properties,
 
 [float]
 ===== Bug fixes
-* Fixed log correlation for log4j2 (#1720)
+* Fixed log correlation for log4j2 - {pull}1720[#1720]
+* Fix apm-log4j1-plugin and apm-log4j2-plugin dependency on slf4j - {pull}1723[#1723]
 
 [[release-notes-1.x]]
 === Java Agent version 1.x

--- a/apm-agent-plugins/apm-grpc/apm-grpc-test-latest/src/test/java/co/elastic/apm/agent/grpc/latest/testapp/generated/HelloGrpc.java
+++ b/apm-agent-plugins/apm-grpc/apm-grpc-test-latest/src/test/java/co/elastic/apm/agent/grpc/latest/testapp/generated/HelloGrpc.java
@@ -29,7 +29,7 @@ import static io.grpc.MethodDescriptor.generateFullMethodName;
 /**
  */
 @javax.annotation.Generated(
-    value = "by gRPC proto compiler (version 1.36.0)",
+    value = "by gRPC proto compiler (version 1.36.1)",
     comments = "Source: rpc.proto")
 public final class HelloGrpc {
 

--- a/apm-agent-plugins/apm-log-shader-plugin/apm-log-shader-plugin-common/src/main/java/co/elastic/apm/agent/log/shader/AbstractLogShadingHelper.java
+++ b/apm-agent-plugins/apm-log-shader-plugin/apm-log-shader-plugin-common/src/main/java/co/elastic/apm/agent/log/shader/AbstractLogShadingHelper.java
@@ -197,12 +197,15 @@ public abstract class AbstractLogShadingHelper<A> {
 
     protected abstract void closeShadeAppender(A shadeAppender);
 
-    /**
-     * Since we shade slf4j in submodules, this provides a way to properly log a message to the agent log.
-     *
-     * @param message message to log
-     */
+    /*********************************************************************************************************************
+     * Since we shade slf4j in submodules, the following methods provide a way to properly log a message to the agent log
+     ********************************************************************************************************************/
+
     protected void logInfo(String message) {
         logger.info(message);
+    }
+
+    protected void logError(String message, Throwable throwable) {
+        logger.error(message, throwable);
     }
 }

--- a/apm-agent-plugins/apm-log-shader-plugin/apm-log4j1-plugin/src/main/java/co/elastic/apm/agent/log4j1/Log4j1LogShadingHelper.java
+++ b/apm-agent-plugins/apm-log-shader-plugin/apm-log4j1-plugin/src/main/java/co/elastic/apm/agent/log4j1/Log4j1LogShadingHelper.java
@@ -31,15 +31,11 @@ import co.elastic.logging.log4j.EcsLayout;
 import org.apache.log4j.FileAppender;
 import org.apache.log4j.RollingFileAppender;
 import org.apache.log4j.WriterAppender;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
 import java.io.IOException;
 
 class Log4j1LogShadingHelper extends AbstractLogShadingHelper<WriterAppender> {
-
-    private static final Logger logger = LoggerFactory.getLogger(Log4j1LogShadingHelper.class);
 
     private static final Log4j1LogShadingHelper INSTANCE = new Log4j1LogShadingHelper();
 
@@ -85,7 +81,7 @@ class Log4j1LogShadingHelper extends AbstractLogShadingHelper<WriterAppender> {
 
                 // todo - set rolling file pattern
             } catch (IOException e) {
-                logger.error("Failed to create Log shading FileAppender. Auto ECS reformatting will not work.", e);
+                logError("Failed to create Log shading FileAppender. Auto ECS reformatting will not work.", e);
             }
         }
         return shadeAppender;

--- a/apm-agent-plugins/apm-log-shader-plugin/apm-log4j2-plugin/src/main/java/co/elastic/apm/agent/log4j2/Log4j2LogShadingHelper.java
+++ b/apm-agent-plugins/apm-log-shader-plugin/apm-log4j2-plugin/src/main/java/co/elastic/apm/agent/log4j2/Log4j2LogShadingHelper.java
@@ -34,16 +34,12 @@ import org.apache.logging.log4j.core.appender.rolling.RolloverStrategy;
 import org.apache.logging.log4j.core.appender.rolling.SizeBasedTriggeringPolicy;
 import org.apache.logging.log4j.core.appender.rolling.TriggeringPolicy;
 import org.apache.logging.log4j.core.config.DefaultConfiguration;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
 import java.lang.reflect.Method;
 
 
 class Log4j2LogShadingHelper extends AbstractLogShadingHelper<AbstractOutputStreamAppender<?>> {
-
-    private static final Logger logger = LoggerFactory.getLogger(Log4j2LogShadingHelper.class);
 
     private static final Log4j2LogShadingHelper INSTANCE = new Log4j2LogShadingHelper();
 
@@ -81,7 +77,7 @@ class Log4j2LogShadingHelper extends AbstractLogShadingHelper<AbstractOutputStre
                 Method getFileNameMethod = originalAppender.getClass().getDeclaredMethod("getFileName");
                 logFile = (String) getFileNameMethod.invoke(originalAppender);
             } catch (Exception e) {
-                logger.error("Failed to obtain log file name from file appender", e);
+                logError("Failed to obtain log file name from file appender", e);
             }
         }
 

